add repository to the credentials prompt

gauron99 opened the pull request:
Added `repository` value (now it is `registry/repository`) to prompt for credentials where its used for printing out tto terminal "during the prompt" to make it easier for the user to see what he is trying to provide credentials for.

Additionally I have created a [new issue](https://github.com/knative/func/issues/2595) that is a followup to this and its' purpose is to further optimize this "credentials" process

fix https://github.com/knative/func/issues/2562

matejvasek commented on pkg/docker/creds/credentials.go:
```diff
@@ -23,9 +23,10 @@ import (
 )
 
-type CredentialsCallback func(registry string) (docker.Credentials, error)
+type CredentialsCallback func(f fn.Function, registry string) (docker.Credentials, error)
```
@gauron99  I do not see any reason for adding `fn.Function` here. The `registry` parameter is completely sufficient.

matejvasek replied:
Just use registry at https://github.com/knative/func/pull/2596/files#diff-753067f7ed9e5f834cd3d8219369a40264abf6ade1665a12e680430f3986c675R27

matejvasek replied:
```go
fmt.Fprintf(out, "Incorrect credentials for registry '%s'. Please make sure the registry is correct and try again.\n", registry)
```

gauron99 replied:
it is so that I can provide the image name and print it out, making it more obvious what are the credentials "used for"

matejvasek replied:
Hmm I thought that registry also included project/namespace something like `docker.io/johndoe`.

gauron99 replied:
I will clean it up a bit, wait. I will ping you when its ready

gauron99 commented:
@matejvasek what about this. I have changed the sole argument that needed it. Added a repository to it so now it is displayed when the prompt asks you for your credentials when trying to 'push' WITH the actual 'registry/repository'
Althought I had to change the prompt tests to reflect this as well

matejvasek commented:
@gauron99 
>what about this. I have changed the sole argument that needed it.

This seems reasonable. Just call it `repository` instead of overly verbose `registryWithRepo`.

IMO that how our flag should have been called: instead of `--registry` we should have had `--repository` or `--img-reposotry`.

gauron99 commented:
@matejvasek @lkingland PTAL

matejvasek commented:
/approve
/lgtm
